packages/core-data/src/actions.js

```diff
@@ -325,15 +325,16 @@ export function* saveEntityRecord(
 				yield receiveAutosaves( persistedRecord.id, updatedRecord );
 			}
 		} else {
-			// Auto drafts should be converted to drafts on explicit saves,
+			// Auto drafts should be converted to drafts on explicit saves and we should not respect their default title,
 			// but some plugins break with this behavior so we can't filter it on the server.
 			let data = record;
-			if (
-				kind === 'postType' &&
-				persistedRecord && persistedRecord.status === 'auto-draft' &&
-				! data.status
-			) {
-				data = { ...data, status: 'draft' };
+			if ( kind === 'postType' && persistedRecord && persistedRecord.status === 'auto-draft' ) {
+				if ( ! data.status ) {
+					data = { ...data, status: 'draft' };
+				}
+				if ( ! data.title || data.title === 'Auto Draft' ) {
```

Comment:
Should Auto Draft be put through a translation function? It is translated in `get_default_post_to_edit()`. Not 100% sure if that is the same place the post title is getting set in Gutenberg.

Reply:
How does that relate to this change? We are clearing the title here.

Reply:
Well it compares the title with an English string, which might not work on a localized environment

Reply:
Got it.

So is "Auto Draft" translated in this context? Where would we check?

Reply:
Yes, as that string is coming from `get_default_post_to_edit()`.

So you'd need to compare it with `__( 'Auto Draft' )`, not just `'Auto Draft'`

Reply:
It looks like it is. I just tried @swissspidy's replication steps on trunk in a non-english locale, and the title was set to `Automatisch gespeicherter Entwurf` on publish.

Reply:
I'm trying to replicate it, but `data.title` is always `undefined` here unless you explicitly typed in the auto draft title, so the second check doesn't even run.